apps/console/src/features/identity-providers/components/forms/authenticators/saml-authenticator-form.tsx

```diff
@@ -0,0 +1,546 @@
+
+    const [ isSLORequestAccepted, setIsSLORequestAccepted ] = useState<boolean>(false);
+    const [ includeProtocolBinding, setIncludeProtocolBinding ] = useState<boolean>(false);
+    const [ isUserIdInClaims, setIsUserIdInClaims ] = useState<boolean>(false);
+    const [ isLogoutEnabled, setIsLogoutEnabled ] = useState<boolean>(false);
+
+    // ISAuthnReqSigned, IsLogoutReqSigned these two fields states will be used by other
+    // fields states. Basically, algorithms fields enable and disable states will be
+    // determine by these two states.
+    const [ isLogoutReqSigned, setIsLogoutReqSigned ] = useState<boolean>(false);
+    const [ isAuthnReqSigned, setIsAuthnReqSigned ] = useState<boolean>(false);
+
+    // This isAlgorithmsEnabled state will control the enable and disable state of
+    // algorithm dropdowns (Signature and Digest)
+    const [ isAlgorithmsEnabled, setIsAlgorithmsEnabled ] = useState<boolean>(false);
+
+    const initialFormValues = useMemo<SamlPropertiesInterface>(() => {
+
+        const [ findPropVal ] = fastSearch(authenticator);
+
+        return {
+            DigestAlgorithm: findPropVal<string>({ defaultValue: "SHA1", key: "DigestAlgorithm" }),
+            IdPEntityId: findPropVal<string>({ defaultValue: "", key: "IdPEntityId" }),
+            NameIDType: findPropVal<string>({ defaultValue: "", key: "NameIDType" }),
+            RequestMethod: findPropVal<string>({ defaultValue: "", key: "RequestMethod" }),
+            SPEntityId: findPropVal<string>({ defaultValue: "", key: "SPEntityId" }),
+            SSOUrl: findPropVal<string>({ defaultValue: "", key: "SSOUrl" }),
+            SignatureAlgorithm: findPropVal<string>({ defaultValue: "RSA with SHA1", key: "SignatureAlgorithm" }),
+            commonAuthQueryParams: findPropVal<string>({ defaultValue: "", key: "commonAuthQueryParams" }),
+            LogoutReqUrl: findPropVal<string>({ defaultValue: "", key: "LogoutReqUrl" }),
+            IncludeProtocolBinding: findPropVal<boolean>({ defaultValue: false, key: "IncludeProtocolBinding" }),
+            ISAuthnReqSigned: findPropVal<boolean>({ defaultValue: false, key: "ISAuthnReqSigned" }),
+            // `IsAuthnRespSigned` is by default set to true when creating the SAML IdP so,
+            // always the value will be true. Keeping this here to indicate for the user and
+            // to enable this if requirements gets changed.
+            IsAuthnRespSigned: findPropVal<boolean>({ defaultValue: true, key: "IsAuthnRespSigned" }),
+            IsLogoutEnabled: findPropVal<boolean>({ defaultValue: false, key: "IsLogoutEnabled" }),
+            IsLogoutReqSigned: findPropVal<boolean>({ defaultValue: false, key: "IsLogoutReqSigned" }),
+            IsSLORequestAccepted: findPropVal<boolean>({ defaultValue: true, key: "IsSLORequestAccepted" }),
+            IsUserIdInClaims: findPropVal<boolean>({ defaultValue: true, key: "IsUserIdInClaims" })
+        } as SamlPropertiesInterface;
```

Comment:
Is it possible to move the key values to constants? It'll look much more readable

Reply:
+1 We could move these objects to `saml-idp-utils` shall we track this separately?